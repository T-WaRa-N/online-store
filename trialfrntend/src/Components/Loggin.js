//importing css atyle
import React from 'react';
import './loggin.css';

// functional component
class Loggin extends React.Component{
    constructor(props){
        super(props)

        //binding the call back function
        this.username = this.username.bind(this)
        this.password = this.password.bind(this)

    }
    
    //callback functions
    username(event){ this.props.username( event.target.value ) }
    password(event){ this.props.password( event.target.value ) }

    render(){
        return (
            <div className="log-in-page">
                <h1>Welcome to the online shopping where you get great discounts on most deals</h1>
                <div className="log-in-form-container">
                    <h3>log-in</h3>
                    <hr/>
                    <form onSubmit={this.props.login}>
                        <label>username:</label><br/>
                        <input type="text" onChange={this.username} /><br/>
                        <label>password:</label><br/>
                        <input type="text" onChange={this.password} /><br/><br/>
                        <button id="log-in-button" type="submit">Log-in</button><br/>
                        <a href="#" onClick={this.props.forgot}>forgot password?</a><br/><br/><hr/>
                    </form>
    
                   <button id="fcbk" onClick={this.props.facebook}>Sign in using Facebook</button><br/><br/>
                   <button id="ggl" onClick={this.props.google}>Sign in using Google</button><br/><br/><hr/>
                   <label>New User</label><br/>
                   <button id="sign-up-button" onClick={this.props.signUp}>Sign-Up</button>
                </div>
                
            </div>
        )
    }
    
}

export default Loggin

