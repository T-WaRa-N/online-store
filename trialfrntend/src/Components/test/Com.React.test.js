import React from 'react';
import Loggin from '../Loggin'
import renderer from 'react-test-renderer'

test(" Loggin page rendered correctly?", ()=>{
    const tree = renderer
    .create(<Loggin />)
    .toJSON()
    expect(tree).toMatchSnapshot()
})
