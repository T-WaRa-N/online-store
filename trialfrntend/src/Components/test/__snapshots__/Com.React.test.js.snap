// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Loggin page rendered correctly? 1`] = `
<div
  className="log-in-page"
>
  <h1>
    Welcome to the online shopping where you get great discounts on most deals
  </h1>
  <div
    className="log-in-form-container"
  >
    <h3>
      log-in
    </h3>
    <hr />
    <form>
      <label>
        username:
      </label>
      <br />
      <input
        onChange={[Function]}
        type="text"
      />
      <br />
      <label>
        password:
      </label>
      <br />
      <input
        onChange={[Function]}
        type="text"
      />
      <br />
      <br />
      <button
        id="log-in-button"
        type="submit"
      >
        Log-in
      </button>
      <br />
      <a
        href="#"
      >
        forgot password?
      </a>
      <br />
      <br />
      <hr />
    </form>
    <button
      id="fcbk"
    >
      Sign in using Facebook
    </button>
    <br />
    <br />
    <button
      id="ggl"
    >
      Sign in using Google
    </button>
    <br />
    <br />
    <hr />
    <label>
      New User
    </label>
    <br />
    <button
      id="sign-up-button"
    >
      Sign-Up
    </button>
  </div>
</div>
`;
